packages/asset-swapper/src/utils/market_operation_utils/fills.ts

```diff
@@ -126,8 +127,8 @@ export function nativeOrdersToFills(
             side === MarketOperation.Sell ? clippedOutput.minus(outputPenalty) : clippedOutput.plus(outputPenalty);
         const adjustedRate =
             side === MarketOperation.Sell ? adjustedOutput.div(clippedInput) : clippedInput.div(adjustedOutput);
-        // Skip orders with rates that are <= 0.
-        if (adjustedRate.lte(0)) {
+        // Optionally skip orders with rates that are <= 0.
+        if (filterNegativeAdjustedRateOrders && adjustedRate.lte(0)) {
```

Comment:
If the rust router chose a native order we want to use it, even if it has negative adjusted rate. In this case it was the best option from all the bad options 🙈

Reply:
do we ever _need_ to skip over orders with rates <= 0?

Reply:
is the assumption here that skipping them saves us some compute time?

Reply:
😫 pain